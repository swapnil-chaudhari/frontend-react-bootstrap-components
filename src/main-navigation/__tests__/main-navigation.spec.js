import React from 'react';
import { Nav, NavItem, NavDropdown, MenuItem } from 'react-bootstrap';
import renderShallow from 'render-shallow';
import MainNavigation from '../main-navigation';

describe('<MainNavigation>', () => {
    describe('when it renders with default logo', () => {
        let component;

        beforeAll(() => {
            component = renderShallow(
                <MainNavigation logoHref="/accounting">
                    <Nav>
                        <NavItem eventKey={ 1 } href="#">Link</NavItem>
                        <NavItem eventKey={ 2 } href="#">Link</NavItem>
                        <NavDropdown eventKey={ 3 } title="Dropdown" id="basic-nav-dropdown">
                            <MenuItem eventKey={ 3.1 }>Action</MenuItem>
                            <MenuItem eventKey={ 3.2 }>Another action</MenuItem>
                            <MenuItem eventKey={ 3.3 }>Something else here</MenuItem>
                            <MenuItem divider />
                            <MenuItem eventKey={ 3.3 }>Separated link</MenuItem>
                        </NavDropdown>
                    </Nav>
                    <Nav pullRight>
                        <NavItem eventKey={ 1 } href="#">Link Right</NavItem>
                        <NavItem eventKey={ 2 } href="#">Link Right</NavItem>
                    </Nav>
                </MainNavigation>
            ).output;
        });

        test('has proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });
    describe('when it renders with custom logo', () => {
        let component;

        beforeAll(() => {
            component = renderShallow(
                <MainNavigation logo={ <div className="custom-logo" /> }>
                    <Nav>
                        <NavItem eventKey={ 1 } href="#">Link</NavItem>
                    </Nav>
                </MainNavigation>
            ).output;
        });

        test('has proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
