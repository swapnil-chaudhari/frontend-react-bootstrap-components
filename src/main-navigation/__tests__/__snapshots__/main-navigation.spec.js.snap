// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainNavigation> when it renders with custom logo has proper markup 1`] = `
<div
  className="MainNavigation"
>
  <Uncontrolled(Navbar)
    bsClass="navigation"
    bsStyle="default"
    className="header-main"
    componentClass="header"
  >
    <NavbarHeader>
      <NavbarBrand>
        <a
          className="logo"
          href="/"
        >
          <div
            className="custom-logo"
          />
        </a>
      </NavbarBrand>
    </NavbarHeader>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={false}
      stacked={false}
    >
      <NavItem
        active={false}
        disabled={false}
        eventKey={1}
        href="#"
      >
        Link
      </NavItem>
    </Nav>
  </Uncontrolled(Navbar)>
</div>
`;

exports[`<MainNavigation> when it renders with default logo has proper markup 1`] = `
<div
  className="MainNavigation"
>
  <Uncontrolled(Navbar)
    bsClass="navigation"
    bsStyle="default"
    className="header-main"
    componentClass="header"
  >
    <NavbarHeader>
      <NavbarBrand>
        <a
          className="logo"
          href="/accounting"
        >
          <Logo />
        </a>
      </NavbarBrand>
    </NavbarHeader>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={false}
      stacked={false}
    >
      <NavItem
        active={false}
        disabled={false}
        eventKey={1}
        href="#"
      >
        Link
      </NavItem>
      <NavItem
        active={false}
        disabled={false}
        eventKey={2}
        href="#"
      >
        Link
      </NavItem>
      <NavDropdown
        eventKey={3}
        id="basic-nav-dropdown"
        title="Dropdown"
      >
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={false}
          eventKey={3.1}
          header={false}
        >
          Action
        </MenuItem>
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={false}
          eventKey={3.2}
          header={false}
        >
          Another action
        </MenuItem>
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={false}
          eventKey={3.3}
          header={false}
        >
          Something else here
        </MenuItem>
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={true}
          header={false}
        />
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={false}
          eventKey={3.3}
          header={false}
        >
          Separated link
        </MenuItem>
      </NavDropdown>
    </Nav>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={true}
      stacked={false}
    >
      <NavItem
        active={false}
        disabled={false}
        eventKey={1}
        href="#"
      >
        Link Right
      </NavItem>
      <NavItem
        active={false}
        disabled={false}
        eventKey={2}
        href="#"
      >
        Link Right
      </NavItem>
    </Nav>
  </Uncontrolled(Navbar)>
</div>
`;
